### ISSUE TITLE
During the training process, the progress bar does not move at 0% when calculating P, R, and mAP

### ISSUE THREAD

**Issue by lemon1022** [ultralytics/yolov5#11339]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

During the training process, the progress bar does not move at 0% when calculating P, R, and mAP

**Comment by glenn-jocher** [comment]
@lemon1022 the progress bar may not move during the calculation of Precision (P), Recall (R), and mean Average Precision (mAP) metrics. This is because these metrics are calculated after each epoch and the progress bar only updates after each batch. Therefore, it is normal for the progress bar to appear to be stuck during the calculation of the aforementioned metrics. You can monitor the progress of the train.py script in the console output which will indicate when the P, R, and mAP values are being calculated.

**Comment by lemon1022** [comment]
![D84P5`G4D)NK@X3{ 1AA_O](https://user-images.githubusercontent.com/88538351/231621636-f87da44c-6881-4d03-a65e-c58761fc0289.png)

always show this status
@glenn-jocher

**Comment by glenn-jocher** [comment]
@lemon1022 The status that you are seeing indicates that the training is still being performed and the calculation of the P, R, and mAP values at the end of each epoch is happening. This message will continue to be displayed until the training is completed. Therefore, you don't have to worry about it and should let the training continue until it has completed. If you do encounter any other issues during the training process or have any further questions, feel free to ask.

**Comment by Haniaaliii** [comment]
I'm training the model on my custom data and i'm getting 0 for the recall & precision values, same for the mAp, I only have 1 class and my current dataset is around 220 images. But i don't think that it's related to the data size because I have a f friend who's working with a Big Data and he has the same issue. Do you know how i could fix this ? 
Thank you,

**Comment by lemon1022** [comment]
@Haniaaliii I have encountered this problem before, and the reason I finally found was the cuda version problem. I dropped the cuda version to 10.2 and it was normal.

**Comment by glenn-jocher** [comment]
@lemon1022 Another possible reason for getting 0 precision and recall during training is the issue of class imbalance, which can occur when you only have a few examples of objects of interest. With only one class and 220 images, you may not have enough samples for the model to learn well. You can consider using data augmentation techniques to create more images or try to collect more data. Additionally, you can try modifying the training parameters, such as adjusting the learning rate or the batch size.

Regarding your friend's issue with a large dataset, it's possible that they may have a similar issue with data imbalance, or it could be related to other factors such as how the data is labeled or the configuration of the model. Without more information, it's difficult to say for sure. However, using a lower version of CUDA, as mentioned by @Haniaaliii, could also be a potential solution to the issue.

If you continue to encounter problems, please provide more details about your model, training parameters, and data, and we can try to assist you further.

**Comment by lemon1022** [comment]
Yes, the model I am training now is because there are too few data (about 100) so that P, R, and mAP are always 0.

**Comment by lemon1022** [comment]
@glenn-jocher  Yes, the model I am training now is because there are too few data (about 100) so that P, R, and mAP are always 0.

**Comment by glenn-jocher** [comment]
@lemon1022, with only 100 images it may be difficult to train a model that gives you good results. Increasing the amount of data you have by collecting more images, or using data augmentation techniques to create more training data, is a good approach to address this issue. When there are few data points, the model overfits and is not able to generalize well to new data, resulting in poor precision and recall.

Another approach you can try is using a pre-trained model as a starting point as it can help to accelerate the training and improve the performance of the model. This can be a good option if you have limited training data. You can fine-tune the pre-trained model on your custom dataset by freezing the backbone layers and only training the prediction head layers.

You can also try using transfer learning where you can use a pre-trained object detection model, such as YOLOv5s or YOLOv5m, and fine-tune it using your data. This can be a good option if you have limited data, as the pre-trained model already has learned general features from a large dataset.

Remember, it is important to choose the right architecture, data and training settings for your problem. Also, take the time to review our documentation to help refine your approach further: https://docs.ultralytics.com/yolov5/.
